### ISSUE TITLE
Scheduled run not working, Oracle Linux 8 UEKR6

### ISSUE THREAD

**Issue by TheNighthawk99** [digint/btrbk#400]
Hello,
I have some issue when running BTRBK as crontab scheduled task.
Please, consider that dryrun option run correctly, as well as a manual one time run.

This is the log:
```
[root@oracle-linux btrfs_pool]# cat /var/log/btrbk.log
2021-06-30T18:45:01+0200 startup v0.31.2 - - - # btrbk command line client, version 0.31.2
2021-06-30T18:45:01+0200 abort_volume ABORT / - - # Failed to fetch subvolume detail
2021-06-30T18:45:01+0200 finished partial - - - # At least one backup task aborted
2021-06-30T19:05:01+0200 startup v0.31.2 - - - # btrbk command line client, version 0.31.2
2021-06-30T19:05:01+0200 abort_volume ABORT /mnt/btrfs_pool - - # Failed to fetch subvolume detail
2021-06-30T19:05:01+0200 finished partial - - - # At least one backup task aborted
2021-06-30T19:07:45+0200 startup v0.31.2 - - - # btrbk command line client, version 0.31.2
2021-06-30T19:07:45+0200 snapshot starting /mnt/btrfs_pool/_snapshots/sys.20210630T1907 /mnt/btrfs_pool/root - -
2021-06-30T19:07:45+0200 snapshot success /mnt/btrfs_pool/_snapshots/sys.20210630T1907 /mnt/btrfs_pool/root - -
2021-06-30T19:07:45+0200 finished success - - - -
```

I have a VM with Oracle Linux 8 UEKR6 installed. This is the fstab file:
```
[root@oracle-linux btrfs_pool]# cat /etc/fstab
UUID=xxxx-xxxx-xxxx-xxxx-x099 /mnt/btrfs_pool         btrfs   subvolid=5,noatime      0 0
UUID=xxxx-xxxx-xxxx-xxxx-x099 /                       btrfs   subvol=root,noatime,ssd,compress=zstd     0 0
UUID=xxxx-xxxx-xxxx-xxxx-x099 /boot                   btrfs   subvol=boot,noatime,ssd,compress=zstd     0 0
UUID=24DB-82E3          /boot/efi               vfat    umask=0077,shortname=winnt 0 2
UUID=xxxx-xxxx-xxxx-xxxx-x099 /home                   btrfs   subvol=home,noatime,ssd,compress=zstd     0 0
UUID=e795995a-6b21-4054-b608-863648593674 none                    swap    defaults        0 0
```

The following is the result of a manual run:
```
[root@oracle-linux btrfs_pool]# btrbk run -v
btrbk command line client, version 0.31.2  (Wed Jun 30 19:07:45 2021)
Using configuration: /etc/btrbk/btrbk.conf
Using transaction log: /var/log/btrbk.log
Creating subvolume snapshot for: /mnt/btrfs_pool/root
[snapshot] source: /mnt/btrfs_pool/root
[snapshot] target: /mnt/btrfs_pool/_snapshots/sys.20210630T1907
Cleaning snapshots: /mnt/btrfs_pool/_snapshots/sys.*
Deleted 0 subvolumes in: /mnt/btrfs_pool/_snapshots/sys.*
Completed within: 0s  (Wed Jun 30 19:07:45 2021)
--------------------------------------------------------------------------------
Backup Summary (btrbk command line client, version 0.31.2)

    Date:   Wed Jun 30 19:07:45 2021
    Config: /etc/btrbk/btrbk.conf

Legend:
    ===  up-to-date subvolume (source snapshot)
    +++  created subvolume (source snapshot)
    ---  deleted subvolume
    ***  received subvolume (non-incremental)
    >>>  received subvolume (incremental)
--------------------------------------------------------------------------------
/mnt/btrfs_pool/root
+++ /mnt/btrfs_pool/_snapshots/sys.20210630T1907
```

The tree command result is below:
```
[root@oracle-linux btrfs_pool]# tree _snapshots/ -L 2
_snapshots/
└── sys.20210630T1907
    ├── bin -> usr/bin
    ├── boot
    ├── dev
    ├── etc
    ├── home
    ├── lib -> usr/lib
    ├── lib64 -> usr/lib64
    ├── media
    ├── mnt
    ├── opt
    ├── proc
    ├── root
    ├── run
    ├── sbin -> usr/sbin
    ├── _snapshots
    ├── srv
    ├── sys
    ├── tmp
    ├── usr
    └── var
```

So two scheduled runs at 18:45 and 19:05 failed, but the manual run at 19:07 performed without hussles.
Do you have any clue how to investigate the matter?

Thank you.
Riccardo

**Comment by digint** [comment]
The transaction log says there was an error "Failed to fetch subvolume detail". I suspect some permission problems, maybe you are not running btrbk as root. Anyways, btrbk would have printed some errors to stderr in this case.
Your cron daemon should have a log file, or send out mail message for failed jobs, including the stderr of the process run (btrbk in this case).

Hope that helps, and sorry for the late reply, I am very busy these days...

**Comment by TheNighthawk99** [comment]
Hello, sorry for the late reply.
Your hint was successful. In my previous test, I had created a crontab for root user **without adding the sudo command**.
I thought that this should be enough.

After reading your message, I tried this command:

`23 04 * * * **sudo** /usr/bin/btrbk run`

I got finally a fully working run based on the schedule.
So, what I note is this:
on Ubuntu server, a crontab task for root user is working **without sudo**;
on CentOS/Oracle Linux server, a crontab task for root user is working **with sudo only**.

I cannot understand why, but this is the fact.
Thank you.
